### PR TITLE
feature: support MXNet 1.4 with MMS

### PR DESCRIPTION
**Pull request by ChoiByungWook**
*Description of changes:*
Add support for MXNet 1.4

The following changes are happening in MXNet 1.4.

Training and serving images are separate.

CUDA updated to 9.2
MXNet updated to MXNet MKL 1.4.0
MXNet-keras updated to 2.2.4.1
Onnx updated to 1.4.1
Python updated to 3.6

The MXNet inference containers now assume the entry_point Python file is stored within /opt/ml/model/code. This is achieved by utilizing the repack_model function.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=61b48f27-34f3-4783-828f-9b1d8907f2f1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=dc4445d4-2696-4919-b443-d23fcd7ed7a1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: doc/using_mxnet.rst [hunk 1]
```diff
@@ -700,6 +700,13 @@ Where ``model`` is the model objected loaded by ``model_fn``, ``request_body`` i
 This one function should handle processing the input, performing a prediction, and processing the output.
 The return object should be one of the following:
 
+For versions 1.4 and higher:
+----------------------------
+- a tuple with two items: the response data and ``accept_type`` (the content type of the response data), or
+- the response data: (the content type of the response will be set to either the accept header in the initial request or a default)
```
Comment (laurenyu) [inline doc/using_mxnet.rst]: what is the default?

File: src/sagemaker/mxnet/README.rst [hunk 2]
```diff
@@ -4,9 +4,9 @@ Using MXNet with the SageMaker Python SDK
 
 With the SageMaker Python SDK, you can train and host MXNet models on Amazon SageMaker.
 
-Supported versions of MXNet: ``1.3.0``, ``1.2.1``, ``1.1.0``, ``1.0.0``, ``0.12.1``.
+Supported versions of MXNet: ``1.4.0``, ``1.3.0``, ``1.2.1``, ``1.1.0``, ``1.0.0``, ``0.12.1``.
 
-Supported versions of MXNet for Elastic Inference: ``1.3.0``.
+Supported versions of MXNet for Elastic Inference: ``1.3.0``, ``1.4.0``.
```
Comment (laurenyu) [inline src/sagemaker/mxnet/README.rst]: I'd keep the order here consistent with the others (i.e. 1.4 first)

File: src/sagemaker/mxnet/README.rst [hunk 3]
```diff
@@ -15,29 +15,30 @@ SageMaker MXNet Containers
...
 
-SageMaker runs MXNet Estimator scripts in either Python 2.7 or Python 3.5. You can select the Python version by passing a ``py_version`` keyword arg to the MXNet Estimator constructor. Setting this to ``py2`` (the default) will cause your training script to be run on Python 2.7. Setting this to ``py3`` will cause your training script to be run on Python 3.5. This Python version applies to both the Training Job, created by fit, and the Endpoint, created by deploy.
+SageMaker runs MXNet Estimator scripts in either Python 2.7 or Python 3.6. You can select the Python version by passing a ``py_version`` keyword arg to the MXNet Estimator constructor. Setting this to ``py2`` (the default) will cause your training script to be run on Python 2.7. Setting this to ``py3`` will cause your training script to be run on Python 3.5. This Python version applies to both the Training Job, created by fit, and the Endpoint, created by deploy.
```
Comment (laurenyu) [inline src/sagemaker/mxnet/README.rst]: * s/MXNet Estimator scripts/MXNet scripts
* s/3.5/3.6 (there's a second one)

File: src/sagemaker/model.py [hunk 4]
```diff
@@ -428,18 +431,28 @@ def prepare_container_def(self, instance_type, accelerator_type=None):  # pylint
...
             self.uploaded_code = None
         else:
-            bucket = self.bucket or self.sagemaker_session.default_bucket()
-            self.uploaded_code = fw_utils.tar_and_upload_dir(session=self.sagemaker_session.boto_session,
-                                                             bucket=bucket,
-                                                             s3_key_prefix=key_prefix,
-                                                             script=self.entry_point,
-                                                             directory=self.source_dir,
-                                                             dependencies=self.dependencies)
+            if repack:
```
Comment (laurenyu) [inline src/sagemaker/model.py]: seems like this part of the conditional should also include `bucket = self.bucket or self.sagemaker_session.default_bucket()` and pass that along to `repack_model()`?

Comment (ChoiByungWook) [reply src/sagemaker/model.py]: out of scope

File: src/sagemaker/mxnet/README.rst (hunk 2 above)
Comment (ChoiByungWook) [reply src/sagemaker/mxnet/README.rst]: oh wow nice catch

File: src/sagemaker/mxnet/README.rst (hunk 3 above)
Comment (ChoiByungWook) [reply src/sagemaker/mxnet/README.rst]: chaned

File: doc/using_mxnet.rst (hunk 1 above)
Comment (ChoiByungWook) [reply doc/using_mxnet.rst]: added

File: src/sagemaker/model.py (hunk 4 above)
Comment (laurenyu) [reply src/sagemaker/model.py]: (talked offline.) fair enough. let's track it internally as a possible enhancement though.

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6fc1d42d-3e9a-4721-b4ec-137b5299696c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker/model.py (hunk 4 above)
Comment (ChoiByungWook) [reply src/sagemaker/model.py]: https://github.com/aws/sagemaker-python-sdk/pull/810

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d7b8b26c-6ab9-468e-a0d9-5769fc09977d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b994e82c-c04f-45a9-924b-4cada0e68df5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**